utils/yoke/src/erased.rs

```diff
@@ -0,0 +1,34 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! This module contains helper types for erasing Cart types.
+//!
+//! See the docs of [`Yoke::erase_rc_cart()`](crate::Yoke::erase_rc_cart)
+//! and [`Yoke::erase_box_cart()`](crate::Yoke::erase_box_cart) for more info.
+//!
+//! Available with the `"alloc"` feature enabled.
+
+use alloc::boxed::Box;
+use alloc::rc::Rc;
+
+/// Dummy trait that lets us `dyn Drop`
+///
+/// `dyn Drop` isn't legal (and doesn't make sense since `Drop` is not
+/// implement on all destructible types). However, all trait objects come with
+/// a destructor, so we can just use an empty trait to get a destructor object.
+pub trait ErasedDestructor: 'static {}
```

Comment:
Thought: We're not actually using this in DataPayload, right?  Because we are instead just Boxing the whole Yoke into a trait object?  So we may or may not need this; if it doesn't solve our use case, is it going to solve anyone else's?

Reply:
Yes, I think that this is generally useful, it's just not useful for _us_ anymore.

It may actually become useful in the future if we ever add e.g. `RcRef` or something.

Reply:
We also can probably clean up the "Box the entire yoke into a trait object" business if we try, actually.